lib/user_agent_parser.ex

```diff
@@ -1,19 +1,25 @@
 defmodule UserAgentParser do
   use Application
 
-  # See http://elixir-lang.org/docs/stable/elixir/Application.html
-  # for more information on OTP Applications
+  alias UserAgentParser.Storage
+
   def start(_type, _args) do
     import Supervisor.Spec, warn: false
 
     children = [
-      # Define workers and child supervisors to be supervised
-      # worker(UserAgentParser.Worker, [arg1, arg2, arg3]),
+      worker(Storage, []),
     ]
 
-    # See http://elixir-lang.org/docs/stable/elixir/Supervisor.html
-    # for other strategies and supported options
     opts = [strategy: :one_for_one, name: UserAgentParser.Supervisor]
     Supervisor.start_link(children, opts)
   end
+
+  def parse(user_agent) do
```

Comment:
This is just a helper, the real work is happening elsewhere.  I personally think these make for a clean public API with `UserAgentPaser.parse("...")` vs something like `UserAgentParser.Parser.parse("...")`.